dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/deaccession/DatasetDeaccessionServiceIT.java

```diff
@@ -0,0 +1,143 @@
+        Dataset dbDataset = datasetService.find(56L);
+        assertEquals(deaccessedVersion.getVersionState(), DatasetVersion.VersionState.DEACCESSIONED);
+        assertTrue(dbDataset.getVersions().contains(deaccessedVersion));
+
+        assertTrue(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.DEACCESSIONED));
+        assertFalse(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.RELEASED));
+        assertTrue(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.ARCHIVED));
+        assertTrue(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.DRAFT));
+    }
+
+    @Test
+    public void shouldDeaccessVersions() {
+        // given
+        Dataset dataset = datasetService.find(56L);
+
+        // when
+        deaccessionService.deaccessVersions(dataset.getVersions(), "TestReason", "https://www.google.com/");
+
+        // then
+        Dataset dbDataset = datasetService.find(56L);
+        for (DatasetVersion version: dbDataset.getVersions()) {
+            assertEquals(DatasetVersion.VersionState.DEACCESSIONED, version.getVersionState());
+        }
+
+        assertTrue(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.DEACCESSIONED));
+        assertFalse(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.RELEASED));
+        assertFalse(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.ARCHIVED));
+        assertFalse(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.DRAFT));
+    }
+
+    @Test
+    public void shouldDeaccessReleasedVersions() {
+        // given
+        Dataset dataset = datasetService.find(56L);
+
+        // when
+        deaccessionService.deaccessReleasedVersions(dataset.getVersions(), "TestReason", "https://www.google.com/");
+
+        // then
+        Dataset dbDataset = datasetService.find(56L);
+        for (DatasetVersion version: dbDataset.getVersions()) {
+            if(version.isDeaccessioned()) {
```

Comment:
This check doesn't make sense :P, isDeaccessioned() already checks the VersionState so there is no way that assertions ever be false.

Reply:
Yeah xD It was that for else clause to verify notequals check. But i agree it cannot stay that way.